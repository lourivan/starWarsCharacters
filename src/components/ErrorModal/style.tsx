import styled from 'styled-components/native'
import { Button } from 'react-native-paper'

export const Container = styled.View`
  flex: 1;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors.text};
`
export const ErrorDescription = styled.View`
  margin: 30px 0;
`
export const ErrorText = styled.Text`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 24px;
`
export const ButtonSkip = styled(Button)`
  margin-top: 40px;
`
