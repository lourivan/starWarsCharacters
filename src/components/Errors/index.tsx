import React from 'react'
import { View } from 'react-native'

// import { Container } from './styles';

export const Errors: React.FC = () => {
  return <View />
}
