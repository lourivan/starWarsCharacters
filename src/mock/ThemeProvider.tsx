import React from 'react'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../'

interface IChildrenProps {
  children: any
}

export const ThemeProviderMock: React.FC<IChildrenProps> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}
