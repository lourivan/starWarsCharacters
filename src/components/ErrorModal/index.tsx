import React from 'react'
import { Modal } from 'react-native'
import LottieView from 'lottie-react-native'

import * as S from './style'

export const ErrorModal = (props) => {
  const { isError, onRetry } = props
  return (
    <Modal transparent={true} animationType={'none'} visible={isError}>
      <S.Container>
        <LottieView
          source={require('assets/lottie/error-animation.json')}
          autoPlay
          loop={false}
          style={{ width: 200, height: 200 }}
        />
        <S.ErrorDescription>
          <S.ErrorText>Oops, Something Went Wrong</S.ErrorText>
          <S.ButtonSkip color={'#000'} onPress={onRetry} mode='contained'>
            Retry
          </S.ButtonSkip>
        </S.ErrorDescription>
      </S.Container>
    </Modal>
  )
}
