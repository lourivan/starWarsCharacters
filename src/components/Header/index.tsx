import React, { useMemo } from 'react'
import { Image } from 'react-native'
import { Appbar } from 'react-native-paper'

import * as S from './style'
import Logo from 'assets/images/Logo.png'
export interface HeaderProps {
  type?: 'default' | 'page'
  onBack?: () => void
  title?: string
  subtitle?: string
}

export const Header: React.FC<HeaderProps> = ({
  type,
  onBack,
  title,
  subtitle,
}) => {
  const TAG = 'Header_component'
  const DefaultHeader = () => (
    <S.Container>
      <S.ImageHeader
        testID={`${TAG}_default_image`}
        source={Image.resolveAssetSource(Logo).uri}
      />
    </S.Container>
  )

  const PageHeader = () => (
    <Appbar.Header>
      <Appbar.BackAction testID={`${TAG}_button_back`} onPress={onBack} />
      <Appbar.Content
        title={title}
        subtitle={subtitle}
        titleStyle={{ fontSize: 20 }}
        subtitleStyle={{ fontSize: 16 }}
      />
    </Appbar.Header>
  )

  const render = useMemo(
    () => (type != 'default' ? <PageHeader /> : <DefaultHeader />),
    [type, title, subtitle, onBack],
  )

  return render
}
