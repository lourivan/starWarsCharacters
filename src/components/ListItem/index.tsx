import React from 'react'

import * as S from './style'
import { assignTestId } from '~/utils/qualityAssurance'

export interface ItemProps {
  title: string
  description?: string
  onPress?: () => void
}

export const ListItem: React.FC<ItemProps> = ({
  title,
  description = '',
  onPress,
}) => {
  const TAG = 'listItem'
  return (
    <S.Item
      {...assignTestId('TouchableOpacity', TAG)}
      title={title}
      description={description}
      right={(_props) => <S.IconRight />}
      onPress={onPress}
    />
  )
}
