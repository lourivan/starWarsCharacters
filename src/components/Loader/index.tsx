import React from 'react'
import { Modal } from 'react-native'
import LottieView from 'lottie-react-native'

import * as S from './style'

export const Loader = (props) => {
  const { loading } = props
  return (
    <Modal transparent={true} animationType={'none'} visible={loading}>
      <S.ModalBackground>
        <LottieView
          source={require('assets/lottie/star-wars.json')}
          autoPlay
          loop
          style={{ width: 200, height: 200 }}
        />
      </S.ModalBackground>
    </Modal>
  )
}
