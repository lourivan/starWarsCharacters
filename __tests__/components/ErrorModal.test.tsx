import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'

import { ErrorModal } from '../../src/components/ErrorModal'
import { ThemeProviderMock } from '../../src/mock/ThemeProvider'

describe('Components/ErrorModal', () => {
  it('renders ErrorModal with default props', async () => {
    const { getByTestId } = render(
      <ThemeProviderMock children={<ErrorModal loading />} />,
    )
    expect(getByTestId('errorModal')).not.toBeNull()
  })
  it('should render snapshot', () => {
    const { toJSON } = render(
      <ThemeProviderMock children={<ErrorModal loading />} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
