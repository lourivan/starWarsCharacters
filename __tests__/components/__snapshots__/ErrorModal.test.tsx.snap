// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ErrorModal should render snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  testID="errorModal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": 0.8571428571428571,
          },
          undefined,
          Object {
            "height": 200,
            "width": 200,
          },
        ]
      }
    >
      <View
        autoPlay={true}
        enableMergePathsAndroidForKitKatAndAbove={false}
        forwardedRef={[Function]}
        loop={false}
        onAnimationFinish={[Function]}
        onLayout={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{\\"v\\":\\"5.0.1\\",\\"fr\\":60,\\"ip\\":0,\\"op\\":67.98,\\"w\\":120,\\"h\\":140,\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ind\\":2,\\"nm\\":\\"Layer 2\\",\\"ks\\":{\\"p\\":{\\"a\\":0,\\"k\\":[59.997,60.37]},\\"a\\":{\\"a\\":0,\\"k\\":[8.95,-21.138,0]},\\"s\\":{\\"a\\":1,\\"k\\":[{\\"t\\":14,\\"s\\":[0,0,100],\\"i\\":{\\"x\\":[0.07],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.86],\\"y\\":[0]},\\"e\\":[92,92,100]},{\\"t\\":63,\\"s\\":[92,92,100]}]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":1,\\"k\\":[{\\"t\\":14,\\"s\\":[0],\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"e\\":[100]},{\\"t\\":63,\\"s\\":[100]}]}},\\"ao\\":0,\\"ip\\":0,\\"op\\":68,\\"st\\":0,\\"bm\\":0,\\"sr\\":1,\\"ty\\":4,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[-0.059,2.139],[0,0],[0,0.147],[2.197,0],[0,-2.285],[0,-0.205],[0,0],[-2.08,0]],\\"o\\":[[2.08,0],[0,0],[0,-0.205],[0,-2.285],[-2.197,0],[0,0.147],[0,0],[0.058,2.139],[0,0]],\\"v\\":[[8.936,-13.535],[12.217,-16.963],[12.598,-38.613],[12.627,-39.17],[8.965,-42.861],[5.273,-39.17],[5.303,-38.613],[5.684,-16.963],[8.936,-13.535]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,2.314],[2.402,0],[0,-2.314],[-2.373,0]],\\"o\\":[[2.402,0],[0,-2.315],[-2.374,0],[0,2.314],[0,0]],\\"v\\":[[8.936,0.586],[13.213,-3.574],[8.936,-7.705],[4.687,-3.574],[8.936,0.586]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.392,0.392,0.392,1]},\\"hd\\":false,\\"o\\":{\\"a\\":0,\\"k\\":100},\\"r\\":1},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0]},\\"s\\":{\\"a\\":0,\\"k\\":[100,100]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"sk\\":{\\"a\\":0,\\"k\\":0},\\"sa\\":{\\"a\\":0,\\"k\\":0}}],\\"nm\\":\\"Object\\",\\"hd\\":false}]},{\\"ind\\":1,\\"nm\\":\\"Layer 1\\",\\"ks\\":{\\"p\\":{\\"a\\":0,\\"k\\":[60,60.18]},\\"a\\":{\\"a\\":0,\\"k\\":[51,51,0]},\\"s\\":{\\"a\\":0,\\"k\\":[104,104,100]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":0,\\"k\\":100}},\\"ao\\":0,\\"ip\\":0,\\"op\\":68,\\"st\\":0,\\"bm\\":0,\\"sr\\":1,\\"ty\\":4,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ty\\":\\"sh\\",\\"d\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[28.167,0],[0,28.167],[-28.166,0],[0,-28.166]],\\"o\\":[[0,28.167],[-28.166,0],[0,-28.166],[29,0],[0,0]],\\"v\\":[[51,0],[0,51],[-51,0],[0,-51],[51,0]],\\"c\\":true},\\"hd\\":false}},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.392,0.392,0.392,1]},\\"o\\":{\\"a\\":0,\\"k\\":100},\\"w\\":{\\"a\\":0,\\"k\\":5},\\"lc\\":1,\\"lj\\":1,\\"ml\\":4},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[51,51]},\\"a\\":{\\"a\\":0,\\"k\\":[0,0]},\\"s\\":{\\"a\\":1,\\"k\\":[{\\"t\\":0,\\"s\\":[0,0],\\"i\\":{\\"x\\":[0.07],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.86],\\"y\\":[0]},\\"e\\":[105,105]},{\\"t\\":49,\\"s\\":[105,105],\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"e\\":[92,92]},{\\"t\\":68,\\"s\\":[92,92]}]},\\"r\\":{\\"a\\":0,\\"k\\":0},\\"o\\":{\\"a\\":1,\\"k\\":[{\\"t\\":0,\\"s\\":[0],\\"i\\":{\\"x\\":[0.833],\\"y\\":[0.833]},\\"o\\":{\\"x\\":[0.167],\\"y\\":[0.167]},\\"e\\":[100]},{\\"t\\":49,\\"s\\":[100]}]},\\"sk\\":{\\"a\\":0,\\"k\\":0},\\"sa\\":{\\"a\\":0,\\"k\\":0}}],\\"nm\\":\\"Object\\",\\"hd\\":false}]}],\\"markers\\":[]}"
        speed={1}
        style={
          Object {
            "aspectRatio": 0.8571428571428571,
            "height": 200,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 30,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
          ]
        }
      >
        Oops, Something Went Wrong
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#000",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "marginTop": 40,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Retry
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
